### PR TITLE
Added apteligent action

### PR DESCRIPTION
**Pull request by Mo7amedFouad**
For more info here https://www.apteligent.com/
docs: https://docs.crittercism.com/ios/ios.html

### CODE REVIEW

File: fastlane/lib/fastlane/actions/apteligent.rb [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+          raise "Couldn't find any dSYM file".red
+        end
+      end
+
+      def self.upload_options(params)
+        file_path = dsym_path(params).shellescape
+
+        options = []
+        options << "--silent"
+        options << "-F dsym=@#{file_path}"
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/apteligent.rb]: What does the `@` do here? Also, can we shellescape the path here too?

File: fastlane/lib/fastlane/actions/apteligent.rb [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        file_path ||= Actions.lane_context[SharedValues::DSYM_OUTPUT_PATH] || ENV[SharedValues::DSYM_OUTPUT_PATH.to_s]
+        file_path ||= Actions.lane_context[SharedValues::DSYM_ZIP_PATH] || ENV[SharedValues::DSYM_ZIP_PATH.to_s]
+
+        if file_path
+          expanded_file_path = File.expand_path(file_path)
+          raise "Couldn't find file at path '#{expanded_file_path}'".red unless File.exist?(expanded_file_path)
+
+          return expanded_file_path
+        else
+          raise "Couldn't find any dSYM file".red
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/apteligent.rb]: Please change this to `UI.user_error!("Couldn't find dSYM file")`

File: fastlane/lib/fastlane/actions/apteligent.rb [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      end
+
+      def self.dsym_path(params)
+        file_path = params[:dsym]
+        file_path ||= Actions.lane_context[SharedValues::DSYM_OUTPUT_PATH] || ENV[SharedValues::DSYM_OUTPUT_PATH.to_s]
+        file_path ||= Actions.lane_context[SharedValues::DSYM_ZIP_PATH] || ENV[SharedValues::DSYM_ZIP_PATH.to_s]
+
+        if file_path
+          expanded_file_path = File.expand_path(file_path)
+          raise "Couldn't find file at path '#{expanded_file_path}'".red unless File.exist?(expanded_file_path)
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/apteligent.rb]: Please change this to `UI.user_error!("Couldn't find ...")`

File: fastlane/lib/fastlane/actions/apteligent.rb [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        # result = Fastlane::Actions.sh(command.join(' '), log: false)
+        shell_command = command.join(' ')
+        result = Helper.is_test? ? shell_command : `#{shell_command}`
+        fail_on_error(result)
+        result
+      end
+
+      def self.fail_on_error(result)
+        if result.include?("error")
+          raise "Server error, failed to upload the dSYM file".red
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/apteligent.rb]: Please change to `UI.crash!("Server error ...")` and also include some error information (contained in `result`)?

File: fastlane/docs/Actions.md [hunk 5]
```diff
@@ -883,6 +883,19 @@ This action allows you to upload symbolication files to Crashlytics. It's extra
 upload_symbols_to_crashlytics(dsym_path: "./App.dSYM.zip")
 ```
 
+### Apteligent
+
+Uploads dSYM.zip file to [Apteligent](https://apteligent.com) for crash symbolication.
+
+```ruby
+apteligent(
+  app_id:'...',
```
Comment (KrauseFx) [inline fastlane/docs/Actions.md]: Please add a space after `:`

**Comment by KrauseFx** [comment]
This looks good, thanks for the PR. Please also add at least one test for the action. I added some comments before it's ready for merging :+1:

File: fastlane/lib/fastlane/actions/apteligent.rb (hunk 1 above)
Comment (Mo7amedFouad) [inline fastlane/lib/fastlane/actions/apteligent.rb]: to indicate that it's a file for curl

File: fastlane/lib/fastlane/actions/apteligent.rb (hunk 2 above)
Comment (Mo7amedFouad) [inline fastlane/lib/fastlane/actions/apteligent.rb]: :+1:

File: fastlane/lib/fastlane/actions/apteligent.rb (hunk 3 above)
Comment (Mo7amedFouad) [inline fastlane/lib/fastlane/actions/apteligent.rb]: :+1:

File: fastlane/lib/fastlane/actions/apteligent.rb (hunk 4 above)
Comment (Mo7amedFouad) [inline fastlane/lib/fastlane/actions/apteligent.rb]: The server return html now with the same message "Server error, failed to upload the dSYM file" :cry: .

File: fastlane/docs/Actions.md (hunk 5 above)
Comment (Mo7amedFouad) [inline fastlane/docs/Actions.md]: :+1:

**Comment by Mo7amedFouad** [comment]
I think I already handled all your comments but I think there is something wrong with `oclint` test cases that keeps failing.

**Comment by KrauseFx** [comment]
Could you rebase from master to fix the tests? Sorry about that

**Comment by asfalcone** [comment]
@Mo7amedFouad Looks like this branch needs to be rebased on master to resolve some conflicts.

**Comment by Mo7amedFouad** [comment]
I couldn't do the rebase correctly. so I'm closing this and opening a new one now. sorry for that

**Pull request closed without merging**